SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -410,17 +417,20 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
 	// Ideally the GlobalStateManager and StaticComponentView would be created as part of USpatialWorkerConnection::Init
 	// however, this causes a crash upon the second instance of running PIE due to a destroyed USpatialNetDriver still being reference.
 	// Why the destroyed USpatialNetDriver is referenced is unknown.
-	if (Connection->GlobalStateManager == nullptr)
+	USpatialGameInstance* GameInstance = GetGameInstance();
+	check(GameInstance != nullptr);
+
+	if (GameInstance->GlobalStateManager == nullptr)
 	{
-		Connection->GlobalStateManager = NewObject<UGlobalStateManager>();
+		GameInstance->GlobalStateManager = NewObject<UGlobalStateManager>();
 	}
-	GlobalStateManager = Connection->GlobalStateManager;
+	GlobalStateManager = GameInstance->GlobalStateManager;
```

Comment:
Could this potentially be restructed where the GameInstance controls the lifetime of the GlobalStateManager and StaticComponentView? As in it creates the objects, and the NetDriver just calls `GetGlobalStateManager()` which returns it?

Reply:
I can move the `NewObject` calls to `USpatialGameInstance::Init`

Reply:
Updated.